File: 1431-Kids-With-the-Greatest-Number-of-Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int maxCandy = 0;
        ArrayList<Boolean> isMostCandy = new ArrayList<>(candies.length);

        for(int candy : candies) {
            maxCandy = Math.max(maxCandy, candy);
        }

        for(int candy : candies) {
            if(candy + extraCandies >= maxCandy) {
                isMostCandy.add(true);
            } else {
                isMostCandy.add(false);
            }
        }

        return isMostCandy;
    }
}

File: 1431-Kids-With-the-Greatest-Number-of-Candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    maxCandies := slices.Max(candies);
    
    isMostCandy := make([]bool, len(candies));
    for i, candy := range candies {
        if candy + extraCandies >= maxCandies {
            isMostCandy[i] = true;
        }
    }

    return isMostCandy;
    }
